src/supervisor/kuberenetes-api-wrappers.ts

```diff
@@ -46,11 +46,15 @@ function shouldRetryRequest(err: IRequestError, attempt: number): boolean {
     return false;
   }
 
-  if (!(err.response)) {
-    return false;
```

Comment:
I think that the way it was before made a bit more sense because if we add another retry-on-status-code (e.g. on 503) then we have to again check if err.response is not falsy. 
```js
  if (err.response && err.response.statusCode === 429) {
    return true;
  }

  if (err.response.statusCode === 503) { // !!! err.response could be undefined
```

And then we'd have to do something like this which just repeats code...
```js
if (err.response && err.response.statusCode === 503) {
```

So something like before would guard us:
```js
  if (!(err.response)) {
    return false;
  }

  if (err.response.statusCode === 429) {
    // safe
  }

  if (err.response.statusCode === 503) {
    // safe
  }
```

Reply:
I changed it because the next errors don't have `response`.  Like this one:
```ts
  if (err.code === 'ETIMEDOUT') {
    return true;
  }
```
However, I agree that won't become pretty. Maybe I can just invert? Include this case first and the case you mentioned after? Do you think it will be better?

Reply:
I think the way it was before was good if we had to add new status codes in the future. I think that won't be the case so let's leave it like it is now with your change 🙏 But if we did need to add a new code, we'll probably change the code the way it was before, so it becomes an unnecessary back-and-forth of the same change 😅